### ISSUE TITLE
Couple of questions

### ISSUE THREAD

**Issue by schatekar** [khelben/IdsvrMultiTenantExample#1]
@khelben 

I am not sure how best to ask you questions, therefore, adding this issue. Let me know if there is a better way to ask questions. Here are my questions

1. Where do you redirect from tenant specific idserver to the central idserver? What happens to tenant-specific cookies on the way?
2. Your implementation of `IClientStore` seems to have hard coded the implementations of client stores for each tenant. Is that how do you do it in practice or this is just for the sake of example? I don't think this is scalable when hosting an organisation wide single instance of id server.

**Comment by BradleyBarnett** [comment]
Agree, @schatekar  I think we would need to just have the tenantID in the route and direct all traffic for each tenant directly into their own setup for IS4, rather than have a "root" IS4.

Our next challenge will be some clients will want to use external auth, such as SAML, and this needs to be configurable on demand for each tenant.  So we will need to dynamically setup new external saml providers under a tenant (can't require code change for this as it would not be scaleable)

**Comment by schatekar** [comment]
@TooMuchTaurine 

I feel that external SSO bit can't be done in a configurable way without refactoring identity server code base. Have you tried anything so far in this area?

**Comment by antonfelich** [comment]
@schatekar 

I'm fairly confident having SSO configurable per tenant with different IdPs is achievable. However where I'm stuck with SAML and IdentityServer4 is that there doesn't appear to be a _pure_ .NET Core library that supports it, which would be required to run in a Linux docker container. 

From what I can tell there are 3 competing "products' available that would suit: -
1. Kentor which has a branch for Core but hasn't been worked on for over 7 months. The current state of the code depends on Framework 4.x so its not pure Core.
2. Elrech which hasn't been worked on for a similar amount of time
3. ComponentSpace who have committed to building a pure Core solution, however they are waiting for Microsoft to migrate the certificate signing over from Framework. 

Have you had any success with SAML and IS4?

**Comment by schatekar** [comment]
@antonfelich

You clearly have more experience with SAML than I do.  I have only used component space, that too outside of identity server.
